### PR TITLE
#22310 Fix `Read database metadata for query analysis` setting enablement state

### LINKED ISSUE #22310: Inconsistent properties for Query Analysis
### Description

Greyed out checkbox becomes togglable after properties reopen

### DBeaver Version

23.3.1

### Operating System

Any

### Database and driver

Any

### Steps to reproduce

1. Open Code editor properties
2. Disable semantic analysis
3. Close and reopen properties

**Expected:** Read database table/columns option is greyed out
**Actual:** It becomes togglable

### Additional context

https://github.com/dbeaver/dbeaver/assets/82543000/7ebd137e-3915-4464-a36e-4139f3a7402c

### PR DESCRIPTION
**Pull request by E1izabeth**
The issue should be fixed

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCodeEditing.java [hunk 1]
```diff
@@ -144,7 +144,8 @@ protected void loadPreferences(DBPPreferenceStore store) {
         csProblemMarkersEnabled.setSelection(store.getBoolean(SQLPreferenceConstants.PROBLEM_MARKERS_ENABLED));
         advancedHighlightingEnabled.setSelection(store.getBoolean(SQLPreferenceConstants.ADVANCED_HIGHLIGHTING_ENABLE));
         readMetadataForSemanticValidationEnabled.setSelection(store.getBoolean(SQLPreferenceConstants.READ_METADATA_FOR_SEMANTIC_ANALYSIS));
-
+        readMetadataForSemanticValidationEnabled.setEnabled(advancedHighlightingEnabled.getSelection());
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCodeEditing.java:147]: So it depends on the `ADVANCED_HIGHLIGHTING_ENABLE`? Why do we have two checkboxes for one option then?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCodeEditing.java:147]: We can use metadata for semantic analysis or now. It's a separate checkbox for that, which should be disabled, if checkbox about semantic analysis is turned off. We have this for a long time, I just fixed synchronization on settings loading.

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCodeEditing.java:147]: I see, it's `setEnabled`

**Review by ShadelessFox — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified

**Pull request merged**